arkouda/client.py

```diff
@@ -523,9 +535,71 @@ def shutdown() -> None:
     serverConfig = None
 
 
+def _json_args_to_str(json_obj: Dict) -> Tuple[int, str]:
+    """
+    Convert Python Dictionary into a JSON formatted string that can be parsed by the msg
+    processing system on the Arkouda Server
+
+    Parameters
+    ----------
+    json_obj : dict
+        Python dictionary of key:val representing command arguments
+
+    Return
+    ------
+    Tuple - the number of parameters found and the json formatted string
+
+    Raises
+    ------
+    TypeError
+        - Keys are a type other than str
+        - Any value is a dictionary.
+        - A list contains values of multiple types.
+
+    Notes
+    -----
+    - Nested dictionaries are not yet supported, but are planned for future support.
+    - Support for lists of pdarray or Strings objects does not yet exist.
+    """
+    from arkouda.pdarrayclass import pdarray
+    from arkouda.strings import Strings
+
+    j = []
+    for key, val in json_obj.items():
+        if not isinstance(key, str):
+            raise TypeError(f"Argument keys are required to be str. Found {type(key)}")
+        if isinstance(val, dict):
+            raise TypeError("Nested JSON is not currently supported for server messages.")
+        param = {"key": key}
+
+        if isinstance(val, pdarray):
+            param["objType"] = "pdarray"
+            param["dtype"] = str(val.dtype)
+            param["val"] = val.name
+        elif isinstance(val, Strings):
```

Comment:
this is cool for now, if it's anticipated we'll get 5+ different types to support here, I recommend converting this if...elif. code block to a dispatch table

Reply:
As of now, we should be able to get away with what is here (though we may add support for nested JSON), but I don't see it exploding. However, with that said, I do agree if it expands further the dispatch table is a better option.

Reply:
cool, works for me